packages/ionic/src/commands/cordova/run.ts

```diff
@@ -234,4 +263,77 @@ ${chalk.cyan('[1]')}: ${chalk.bold('https://ionicframework.com/docs/developer-re
       await this.runCordova(filterArgumentsForCordova(metadata, options));
     }
   }
+
+  protected async nativeRun(args: any): Promise<void> {
+    if (!this.project) {
+      throw new FatalException(`Cannot run ${chalk.green('ionic cordova run/emulate')} outside a project directory.`);
+    }
+
+    const p = await this.env.shell.spawn('native-run', args, { stdio: 'pipe', cwd: this.project.directory });
+
+    // no resolve, native-run --connect stays running until SIGINT or app close
+    return new Promise<void>((resolve, reject) => {
+      const closeHandler = async (code: number | null) => {
+        if (code !== null && code !== 0) { // tslint:disable-line:no-null-keyword
+          if (code === -2) {
```

Comment:
I was hoping you would know! I dug in a little and it appears to be the value of `UV_ENOENT` https://github.com/nodejs/node/blob/ad6ead3aab24e6b703d4982ba0563a15561a2915/lib/internal/child_process.js#L353.

It seems negative error codes is the `libuv` convention: http://docs.libuv.org/en/v1.x/errors.html.

The error handler receives the regular node ENOENT error before the close handler though, so I've moved this there and now ignore UV_ENOENT in the close handler to prevent double scary messages.